### ISSUE TITLE
context menu does not have an `Add to Photos` option in certain cases

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-ios#2011]
### Description: 
Sometimes the context menu does not have a Save option when trying to save an image or a gif. It seems that if you are trying to save the image/gif itself, then the context menu does have a Save option. However, if the image/gif has a hyperlink then the context menu does not have a Save option.

Both Safari and Firefox correctly handle this and show a Save option whether there is a hyperlink or not.

### Steps to Reproduce 
  1. Navigate to a site with gifs or images. (I used giphy.com)
  2. Navigate to https://giphy.com/gifs/Friends-episode-15-friends-tv-the-one-where-estelle-dies-W3a0zO282fuBpsqqyD and long press on the gif.
  3. Context menu displays an `Add to Photos` option.
  4. Navigate to https://giphy.com/gifs/ghostbustersmovies-ghostbusters-2-3o72F7KsTuiBXvvDLG
  5. Long press on the gif.

**Actual result:** 
No `Save to Photos` option since there is a hyperlink to a landing page.
![contextmenu1-14](https://user-images.githubusercontent.com/28145373/69377389-2f759580-0c7a-11ea-8eaa-a5ee01b474e0.jpeg)

**Expected result:**
Some `Save to Photos` option like first gif you tried:
![Image from iOS](https://user-images.githubusercontent.com/28145373/69377362-1967d500-0c7a-11ea-9f75-ef76ea491d62.png)

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
easily

**Brave Version:** 
1.13 and 1.14

**Device details:** 
iPad Mini iOS13

**Website problems only:**
- did you check with Brave Shields down? n/a
- did you check in Safari/Firefox (WkWebView-based browsers)?  yes, works on both Safari and FF:
![contextmenu-safari](https://user-images.githubusercontent.com/28145373/69377469-5df37080-0c7a-11ea-92bb-8ebbe279128f.jpeg)
![contextmenu-FF](https://user-images.githubusercontent.com/28145373/69377474-60ee6100-0c7a-11ea-8073-0feaad25753a.jpeg)

### Additional Information
cc @srirambv

**Comment by Brave-Matt** [comment]
Hello --  I have a user reporting this behavior and I'm also able to reproduce it. This issue is very old but is there a particular reason it hasn't been touched since opened? cc @LaurenWags @srirambv @anthonypkeane 

User report: https://community.brave.com/t/cannot-save-photos/216851

**Comment by RonanCodes** [comment]
I see the context option after holding my finger down.
But the photo doesn't actually save.
This is on my iPad.

This link for example:
https://mapofeurope.com/wp-content/uploads/2018/03/europe-political-map.jpg

![image](https://user-images.githubusercontent.com/6930753/159178787-aaabca43-83cc-487a-ab59-f5afd186b825.jpeg)
